source/dub/packagemanager.d

```diff
@@ -377,21 +377,42 @@ class PackageManager {
 		Params:
 			path = NativePath to the root directory of the package
 			recipe_path = Optional path to the recipe file of the package
-			allow_sub_packages = Also return a sub package if it resides in the given folder
+			name = Optional (sub-)package name if known in advance. Required if a sub-package is to be returned.
 			mode = Whether to issue errors, warning, or ignore unknown keys in dub.json
 
 		Returns: The packages loaded from the given path
 		Throws: Throws an exception if no package can be loaded
 	*/
 	Package getOrLoadPackage(NativePath path, NativePath recipe_path = NativePath.init,
-		bool allow_sub_packages = false, StrictMode mode = StrictMode.Ignore)
+		PackageName name = PackageName.init, StrictMode mode = StrictMode.Ignore)
```

Comment:
The breaking API change is in a separate 2nd commit, in case we cannot/don't want to break it.